specification/dnc/resource-manager/Microsoft.DelegatedNetwork/preview/2020-08-08-preview/DelegatedNetwork.json

```diff
@@ -0,0 +1,820 @@
+          "200": {
+            "description": "OK."
+          },
+          "202": {
+            "description": "Accepted."
+          },
+          "204": {
+            "description": "No Content."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworks/{vnetName}/subnets/{subnetName}/providers/Microsoft.DelegatedNetwork/delegatedSubnet": {
+      "get": {
+        "description": "Gets details about the specified dnc DelegatedSubnet Link.",
+        "x-ms-examples": {
+          "Get details of a delegated subnet": {
+            "$ref": "./examples/getDelegatedSubnet.json"
+          }
+        },
+        "tags": [
+          "DNCDelegatedSubnet"
+        ],
+        "operationId": "DelegatedSubnet_GetDetails",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/vnetNameParameter"
+          },
+          {
+            "$ref": "#/parameters/subnetNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The operation was successful.",
+            "schema": {
+              "$ref": "#/definitions/DelegatedSubnet"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "description": "Put delegated subnet resource",
+        "x-ms-examples": {
+          "put delegated subnet": {
+            "$ref": "./examples/putDelegatedSubnet.json"
+          }
+        },
+        "tags": [
+          "DNCDelegatedSubnet"
+        ],
+        "operationId": "DelegatedSubnet_PutDetails",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/vnetNameParameter"
+          },
+          {
+            "$ref": "#/parameters/subnetNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "delegatedSubnetParameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DelegatedSubnetParameters"
```

Comment:
request body and response body must use the same model.

Reply:
why is that requirement? response body can have state saying succeeded or failed and request body doesn't need that.

Reply:
i have seen many existing RPs using different request and response body